### PR TITLE
Postgres storage: remove id/last_modified from data field (fixes #762)

### LINKED ISSUE #762: Postgresql storage: id (and last_modified) are stored in the data column
We could «save» some storage space by avoiding this. It is rather trivial to do: strip record of those fields before storing it as `data`.

Current insert query is:
```sql
	        WITH delete_potential_tombstone AS (
	            DELETE FROM deleted
	             WHERE id = '861756c7-9405-4819-a2cb-336606d3540e'
	               AND parent_id = '1234'
	               AND collection_id = 'test'
	        )
	        INSERT INTO records (id, parent_id, collection_id, data, last_modified)
	        VALUES ('861756c7-9405-4819-a2cb-336606d3540e', '1234',
	                'test', ('{"code":{"sub":"c"},"id":"861756c7-9405-4819-a2cb-336606d3540e"}')::JSONB,
	                from_epoch(NULL))
	        RETURNING id, as_epoch(last_modified) AS last_modified;

```

### PR DESCRIPTION
**Pull request by gabisurita**
Fixes #762 

- [x] Add a changelog entry.
- [x] Add your name in the contributors file.

### CODE REVIEW

File: kinto/core/storage/postgresql/__init__.py [hunk 1]
```diff
@@ -230,6 +230,12 @@ def create(self, collection_id, parent_id, record, id_generator=None,
         else:
             record[id_field] = id_generator()
 
+        # Remove redundancy in data field
+        query_record = record.copy()
+        del query_record[id_field]
+        if modified_field in record:
+            del query_record[modified_field]
```
Comment (leplatrem) [inline kinto/core/storage/postgresql/__init__.py]: 👍 

You could even make it shorter like this:
```
query_record.pop(id_field, None)
query_record.pop(modified_field, None)
```

**Comment by leplatrem** [comment]
Thanks for this contribution!

We should also do it for `update()`.

Since this applies to the postgresql backend internals, it would exceptionnally acceptable to merge it without tests. But maybe it would be relevant to have a test in `storage/testing.py` that asserts that the specified `record` object is not altered (that's why you added the line with `.copy()` I suppose). What do you think ?

**Comment by gabisurita** [comment]
Removing the id field already leads to an error on the insertion, but the removing the timestamp doesn't. I think including a test for it would be a good idea. I'll work on it.

**Comment by leplatrem** [comment]
The current code looks great and seems to do the job! Adding a test would make it perfect :)
(*do not hesitate to ask for help if you're stuck*)

**Comment by gabisurita** [comment]
Sorry for the delay, I was having some troubles in get the storage tests running locally. Some tests were failing due to the database timezone. This probably should be included on the backend documentation, but I think that's already covered by issue #788. 

Leaving the last_modified field off the record already produce errors on several tests, so I think another test for this is not needed.
`
Results (159.86s):
    1386 passed
       6 failed
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:567: AssertionError: 1475004498137L not less than 1475004498124L
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:548: AssertionError: 1475004498292L != 1448881675541
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:498: AssertionError: 100 != 200
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:650: KeyError: 'last_modified'
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:629: AssertionError: 1475004507708L not less than 1475004507701L
         - /home/gabi/outreachy/kinto/kinto/core/storage/testing.py:610: KeyError: 'last_modified'
`

**Comment by leplatrem** [comment]
Well, it's always better to be able to run the tests locally... 

I pulled the branch, and the whole test suite is green indeed. And you're right there is a test called `test_create_copies_the_record_before_modifying_it` in the storage tests.

So I think we're good to merge!

Thanks for the good work!

**Pull request merged**